bench-tps/src/bench.rs

```diff
@@ -76,11 +76,28 @@ struct KeypairChunks<'a> {
 impl<'a> KeypairChunks<'a> {
     /// Split input vector of keypairs into two sets of chunks of given size
     fn new(keypairs: &'a [Keypair], chunk_size: usize) -> Self {
+        // Use `chunk_size` as the number of conflict groups per chunk so that each destination key is unique
+        Self::new_with_conflict_groups(keypairs, chunk_size, chunk_size)
+    }
+
+    /// Split input vector of keypairs into two sets of chunks of given size. Each chunk
+    /// has a set of source keys and a set of destination keys. There will be
+    /// `num_conflict_groups_per_chunk` unique destination keys per chunk, so that the
+    /// destination keys may conflict with each other.
+    fn new_with_conflict_groups(
+        keypairs: &'a [Keypair],
+        chunk_size: usize,
+        num_conflict_groups_per_chunk: usize,
+    ) -> Self {
         let mut source_keypair_chunks: Vec<Vec<&Keypair>> = Vec::new();
         let mut dest_keypair_chunks: Vec<VecDeque<&Keypair>> = Vec::new();
         for chunk in keypairs.chunks_exact(2 * chunk_size) {
             source_keypair_chunks.push(chunk[..chunk_size].iter().collect());
-            dest_keypair_chunks.push(chunk[chunk_size..].iter().collect());
+            dest_keypair_chunks.push(
+                (chunk_size..2 * chunk_size)
+                    .map(|idx| &chunk[idx % num_conflict_groups_per_chunk])
+                    .collect(),
+            );
```

Comment:
Probably correct, but I kind of don't get it.
Assume `chunk_size = num_conflict_groups_per_chunk = 2`:
```python
 chunk = [x1, x2, x3, x4]
 source_keypair_chunk[0] = [&x1, &x2]
 dest_keypair_chunks[0] = [&chunk[2%2], &chunk[3%2]] = [&x1, &x2] // <- shouldn't it be [&x3, &x4]?
 ```

Reply:
Nope, not correct. I just screwed up the code when I was "cleaning up".
You're example is correct, it should be `[&x3, &x4]`

Reply:
I fixed this, and added test for `KeyChunks::new` and `KeyChunks::new_with_conflict_groups`